libs/common/src/main/java/org/opensearch/common/Booleans.java

```diff
@@ -45,30 +45,72 @@ private Booleans() {
     /**
      * Parses a char[] representation of a boolean value to <code>boolean</code>.
      *
-     * @return <code>true</code> iff the sequence of chars is "true", <code>false</code> iff the sequence of chars is "false" or the
-     * provided default value iff either text is <code>null</code> or length == 0.
+     * @return <code>true</code> iff the sequence of chars is "true", <code>false</code> iff the sequence of
+     * chars is "false" or the provided default value iff either text is <code>null</code> or length == 0.
      * @throws IllegalArgumentException if the string cannot be parsed to boolean.
      */
     public static boolean parseBoolean(char[] text, int offset, int length, boolean defaultValue) {
-        if (text == null || length == 0) {
+        if (text == null) {
             return defaultValue;
-        } else {
-            return parseBoolean(new String(text, offset, length));
         }
+
+        switch (length) {
+            case 0:
+                return defaultValue;
+            case 1:
+            case 2:
+            case 3:
```

Comment:
Keeping these cases on purpose to ensure the switch is dense (no holes) in order to guarantee this gets converted to a [tableswitch instruction](https://stackoverflow.com/questions/10287700/difference-between-jvms-lookupswitch-and-tableswitch), which is _really_ fast. Most likely the compiler already fills such small holes.